velox/dwio/parquet/reader/PageReader.cpp

```diff
@@ -368,6 +370,32 @@ void PageReader::prepareDictionary(const PageHeader& pageHeader) {
           // We start from the end to allow in-place expansion.
           values[i] = parquetValues[i];
         }
+      } else if (
+          type_->type()->isTimestamp() && parquetType == thrift::Type::INT64) {
+        VELOX_CHECK(type_->logicalType_.has_value());
+        auto logicalType = type_->logicalType_.value();
+        if (logicalType.__isset.TIMESTAMP) {
+          if (!logicalType.TIMESTAMP.isAdjustedToUTC) {
+            VELOX_NYI("Only UTC adjusted Timestamp is supported.");
```

Comment:
Hi @mskapilks. After your PR is merged, I will submit a follow-up PR to support the case when `isAdjustedToUTC=true` because I have already added the timezone information to the Parquet reader.

Reply:
@liujiayi771 Thats great, thanks :+1: